src/Microsoft.Data/StringColumn.BinaryOperations.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    public partial class StringColumn : BaseColumn
+    {
+        public override BaseColumn Add(BaseColumn column)
+        {
+            // TODO: Using indexing is VERY inefficient here. Each indexer call will find the "right" buffer and then return the value
+            if (Length != column.Length)
+            {
+                throw new ArgumentException($"Column lengths are mismatched", nameof(column));
+            }
+            StringColumn ret = Clone();
+            for (long i = 0; i < Length; i++)
+            {
+                ret[i] += column[i].ToString();
+            }
+            return ret;
+        }
+
+        public override BaseColumn Add<T>(T value)
+        {
+            StringColumn ret = Clone();
+            string valString = value.ToString();
+            for (int i = 0; i < ret.stringBuffers.Count; i++)
+            {
+                IList<string> buffer = ret.stringBuffers[i];
+                int bufferLen = buffer.Count;
+                for (int j = 0; j < bufferLen; j++)
+                {
+                    buffer[j] += valString;
+                }
+            }
+            return ret;
+        }
+
+        public override BaseColumn Equals(BaseColumn column)
+        {
+            // TODO: Using indexing is VERY inefficient here. Each indexer call will find the "right" buffer and then return the value
+            if (Length != column.Length)
+            {
+                throw new ArgumentException($"Column lengths are mismatched", nameof(column));
+            }
+            PrimitiveColumn<bool> ret = new PrimitiveColumn<bool>(Name);
+            for (long i = 0; i < Length; i++)
+            {
+                ret.Append(this[i] == column[i]);
+            }
+            return ret;
+        }
+
+        public override BaseColumn Equals<T>(T value)
+        {
+            PrimitiveColumn<bool> ret = new PrimitiveColumn<bool>(Name);
+            string valString = value.ToString();
+            for (int i = 0; i < stringBuffers.Count; i++)
+            {
+                IList<string> buffer = stringBuffers[i];
+                int bufferLen = buffer.Count;
+                for (int j = 0; j < bufferLen; j++)
```

Comment:
Why does the `Equals` method above use a single `for` loop over `Length` and the index, but here we are doing 2 `for` loops over the `stringBuffers`?

Reply:
I wrote that initially thinking the 2 for loop version may be more optimized since I'm avoiding indexing. I changed it now to use the single for loop though. When I fix https://github.com/dotnet/corefxlab/issues/2659 though, the single loop version should be super efficient.